src/EmailAccountRecovery.sol

```diff
@@ -0,0 +1,217 @@
+                keccak256(
+                    abi.encodePacked(
+                        type(ERC1967Proxy).creationCode,
+                        abi.encode(
+                            guardianVerifierImplementation,
+                            abi.encodeCall(
+                                IGuardianVerifier.initialize,
+                                (
+                                    recoveredAccount,
+                                    accountSalt,
+                                    verifierInitData
+                                )
+                            )
+                        )
+                    )
+                )
+            );
+    }
+
+    /// @notice Deploys a new proxy contract for guardian verification.
+    /// @dev This function uses the CREATE2 opcode to deploy a new ERC1967Proxy contract with a
+    /// deterministic address.
+    /// @param guardianVerifierImplementation The address of the guardian verifier implementation.
+    /// @param accountSalt A bytes32 salt value used to ensure the uniqueness of the deployed proxy address.
+    /// @param recoveredAccount The address of the account to be recovered.
+    /// @param verifierInitData The initialization data for the guardian verifier.
+    /// @return address The address of the newly deployed proxy contract.
+    function deployGuardianVerifierProxy(
+        address guardianVerifierImplementation,
+        address recoveredAccount,
+        bytes32 accountSalt,
+        bytes memory verifierInitData
+    ) internal virtual returns (address) {
+        if (
+            guardianVerifierImplementation == address(0) ||
+            guardianVerifierImplementation.code.length == 0
+        ) {
+            revert InvalidGuardianImplementation();
+        }
+
+        ERC1967Proxy proxy = new ERC1967Proxy{salt: accountSalt}(
+            guardianVerifierImplementation,
+            abi.encodeCall(
+                IGuardianVerifier.initialize,
+                (recoveredAccount, accountSalt, verifierInitData)
+            )
+        );
+        return address(proxy);
+    }
+
+    /// @notice Handles an acceptance by a new guardian.
+    /// @dev This function validates the email auth message, deploys a new EmailAuth contract as a
+    /// proxy if validations pass and initializes the contract.
+    /// @param guardianVerifierImplementation The address of the guardian verifier implementation.
+    /// @param recoveredAccount The address of the account to be recovered.
+    /// @param accountSalt A bytes32 salt value used to ensure the uniqueness of the deployed guardian verifier.
+    /// @param verifierInitData The initialization data for the guardian verifier.
+    /// @param proofData The proof data required for the guardian verifier.
+    /// the command in the given email auth message.
+    function handleAcceptance(
+        address guardianVerifierImplementation,
+        address recoveredAccount,
+        bytes32 accountSalt,
```

Comment:
One small idea to simplify the interface is to pass the accountSalt as part of the proofData parameter. It doesn't matter too much for the verifiers we want to support for the grant scope as believe they would all use account code, but for additional non-zkemail zk verifiers, this would change

Reply:
Need to keep accountSalt separately as it's the salt to the create2 address of the new guardian-verifier as well, `computeGuardianVerifierAddress`. I just think it keep consistency in the functions if this arg is separate.

Reply:
ah yeah true. Let's keep it as is.

We will look to make this interface more generic for our v2, which would include making the create2 values generic, or separating the guardian deployment and handleAcceptance calls, but that can be out of scope for this work.